### PR TITLE
Bom 1201

### PR DESCRIPTION
**Pull request by Ayub-Khan**
-Added pytest and pytest-django in requirements
-Updated requirements
-remove django-nose settings

### CODE REVIEW

**Comment by Ayub-Khan** [comment]
@jmbowman A lot of tests are failing because of some db error which im not being able to reproduce locally. Any idea ?

**Comment by jmbowman** [comment]
My first guess would be that the randomization of the test sequence via pytest-randomly is uncovering bugs in test setup and teardown; I think the tests were running in a deterministic order under nose.  We should eventually fix those bugs, but for now try adding ` -p no:randomly` to the `addopts` pytest setting.  You should see randomly disappear from the plugin list in the pytest output:

```plugins: randomly-3.2.1, metadata-1.8.0, django-webtest-1.9.7, timeout-1.3.4, base-url-1.4.1, variables-1.9.0, django-3.8.0, html-1.22.1, selenium-1.17.0```

**Comment by Ayub-Khan** [comment]
@jmbowman pytest-django-ordering fixed the issues.

File: requirements/test.in [hunk 1]
```diff
@@ -16,9 +15,12 @@ isort
 lxml
 mock
 mock-django
-nose-ignore-docstring
 pycodestyle
 pylint
+pytest
+pytest-cov
+pytest-django
+pytest-django-ordering
```
Comment (jmbowman) [inline requirements/test.in:23]: Nice catch, I'll mention this in future nose-to-pytest conversion tickets.

**Review by awais786 — APPROVED** [review]

File: tox.ini [hunk 2]
```diff
@@ -66,9 +71,7 @@ commands =
     detect_changed_translations: i18n_tool changed
     validate_translations: i18n_tool validate -
 
-    tests: coverage run --branch --source=ecommerce \
-    tests:     ./manage.py test {posargs:ecommerce} \
-    tests:     --settings=ecommerce.settings.test --with-ignore-docstrings --logging-level=DEBUG
+    tests: pytest --cov=ecommerce --cov-report term --cov-config=.coveragerc --no-cov-on-fail --ds=ecommerce.settings.test -p no:randomly
```
Comment (jmbowman) [inline tox.ini]: Do you need `--ds=ecommerce.settings.test` here even with `DJANGO_SETTINGS_MODULE` configured under `[pytest]` above?  And is `-p no:randomly` still needed now that we're using pytest-django-ordering?

**Review by jmbowman — APPROVED** [review]

File: tox.ini (hunk 2 above)
Comment (Ayub-Khan) [reply tox.ini]: we still need randomly as that's being used in e2e tests.

**Review by jmbowman — APPROVED** [review]

**Pull request merged**